php-frontend/src/main/java/org/sonar/php/PHPAnalyzer.java

```diff
@@ -105,10 +110,16 @@ public List<PhpIssue> analyze() {
       });
       allIssues.addAll(issues);
     }
+    if (suppressWarningFilter != null) {
+      suppressWarningFilter.scanCompilationUnit(currentFile.uri().toString(), currentFileTree);
```

Comment:
I'm not sure if this is possible, but wouldn't it be great if you get the `currentFile.uri()` from the `context`?
Than you could adopt the SupressWarningFilter to be more consistent with other `PHPCheck`s using the `analyze` and you wouldn't have to create a method `scanCompilationUnit` which sounds exactly like the `visitCompilationUnit` method.

Reply:
Seems like a good idea! Will check this, sometimes the context is null and I'm not sure why, but otherwise sounds like the right way to proceed.